src/rust/ide/src/controller/module.rs

```diff
@@ -25,13 +42,56 @@ use crate::double_representation::definition::DefinitionName;
 // ============
 
 /// Path identifying module's file in the Language Server.
-pub type Path = language_server::Path;
+#[derive(Clone,Debug,Eq,Hash,PartialEq)]
+pub struct Path {
+    file_path : FilePath,
+}
+
+impl Path {
+    /// Create a path from the file path. Returns None if given path is not a valid module file.
+    pub fn from_file_path(file_path:FilePath) -> Option<Self> {
+        let has_proper_ext   = file_path.extension() == Some(constants::LANGUAGE_FILE_EXTENSION);
+        let first_char       = file_path.file_name().and_then(|name| name.chars().next());
+        let capitalized_name = first_char.map_or(false, |ch| ch.is_uppercase());
+        let is_module        = has_proper_ext && capitalized_name;
+        is_module.and_option_from(|| Some(Path{file_path}))
```

Comment:
Here, for every call that returns `Option`, I'd apply `?` operator immediately.
If there is no extension or no characters in name, then we should early return, rather than accumulating false flags.